lib/elixir/src/elixir_scope.erl

```diff
@@ -75,6 +76,20 @@ warn_underscored_var(Line, File, Name, Kind) ->
       ok
   end.
 
+warn_underscored_var_access(Line, File, Name) ->
+  case atom_to_list(Name) of
+    "__impl__" ->
+      ok; %% Variable needed for protocols
+    "_rewrite" ->
```

Comment:
We should fix those. We probably don't need to name it `_rewrite` nor `__impl__`. :)